packages/wrangler/src/index.ts

```diff
@@ -1021,15 +1062,46 @@ export async function main(argv: string[]): Promise<void> {
 		recordedCommand = true;
 		// `args._` doesn't include any positional arguments (e.g. script name,
 		// key to fetch) or flags
-		addBreadcrumb(`wrangler ${args._.join(" ")}`);
+
+		try {
+			const { rawConfig, configPath } = parseConfig(args.config, args, false);
+			dispatcher = getMetricsDispatcher({
+				sendMetrics: rawConfig.send_metrics,
+				configPath,
+			});
+		} catch (e) {
+			// If we can't parse the config, we can't send metrics
+			logger.debug("Failed to parse config. Disabling metrics dispatcher.", e);
+		}
+
+		command = `wrangler ${args._.join(" ")}`;
+		metricsArgs = args;
+		addBreadcrumb(command);
+		// NB despite 'applyBeforeValidation = true', this runs *after* yargs 'validates' options,
+		// e.g. if a required arg is missing, yargs will error out before we send any events :/
+		void dispatcher?.sendNewEvent("wrangler command started", {
+			command,
+			args,
+		});
 	}, /* applyBeforeValidation */ true);
 
 	let cliHandlerThrew = false;
 	try {
 		await wranglerWithMiddleware.parse();
+
+		const durationMs = Date.now() - startTime;
+
+		void dispatcher?.sendNewEvent("wrangler command completed", {
+			command,
+			args: metricsArgs,
```

Comment:
Why does the completed event need the args?

Reply:
there are some cases where we are just looking at the completed event where we also need the args  (e.g. the duration of wrangler dev)